lib/temporal/client.rb

```diff
@@ -281,9 +281,13 @@ def await_workflow_result(workflow, workflow_id:, run_id: nil, timeout: nil, nam
     # @param workflow_task_id [Integer, nil] A specific event ID to reset to. The event has to
     #   be of a type WorkflowTaskCompleted, WorkflowTaskFailed or WorkflowTaskTimedOut
     # @param reason [String] a reset reason to be recorded in workflow's history for reference
+    # @param request_id [String, nil] an idempotency key for the Reset request or `nil` to use
+    #   an auto-generated, unique value
+    # @param reset_reapply_type [Symbol, nil] one of the Temporal::ResetReapplyType values or `nil`
```

Comment:
I assume this maps to the unspecified enum value? Does Temporal make any claim about how that will behave? You might instead pick `None` as the default and require a valid enum value here.

Reply:
Exactly, yeah it maps to unspecified.

There aren't any claims about it in the gRPC API contract, and to be honest there are practically no docs on the semantics of Reset (that I could find).

Looking at the server implementation it looks like this maps to `:signal` as the default ([source](https://github.com/yiminc/temporal/blob/30a89202af4f508def300c4ae6232f1045be90a1/common/enums/defaults.go#L67)), so I'm inclined to make that the default.